pkg/storage/internalstorage/register.go

```diff
@@ -93,11 +93,12 @@ func NewStorageFactory(configPath string) (storage.StorageFactory, error) {
 	sqlDB.SetMaxOpenConns(connPool.MaxOpenConns)
 	sqlDB.SetConnMaxLifetime(connPool.ConnMaxLifetime)
 
-	if err := db.AutoMigrate(&Resource{}); err != nil {
```

Comment:
I think you can determine DiviionPolicy and SkipAutoMigration first to decide whether to auto-migrate here or not, so that you can avoid querying if the table exists every time you call ResourceStorage